### PR TITLE
Code coverage

### PR DESCRIPTION
**Pull request by rrudolfo**
I have created test files and edited to some of the pre-existing to get 90% code coverage. The files to be reviewed are, UMLMethods, UMLClass, StateManager, Relationship, and RelationshipType tests.

### CODE REVIEW

File: src/main/java/codecain/BackendCode/Model/RelationshipType.java [hunk 1]
```diff
@@ -35,6 +35,10 @@ public String toString(){
     }
 
     public static RelationshipType fromString(String type){
+        if (type == null || type.trim().isEmpty()){
+            System.err.println("Input cannot be null or empty.");
```
Comment (colbycrutcher) [inline src/main/java/codecain/BackendCode/Model/RelationshipType.java:39]: good error handling

File: src/test/java/RelationshipTypeTests.java [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    }
+
+    /**
+     * Tests the {@code getArrowString()} method of {@link RelationshipType}.
+     * Verifies that the method returns the correct arrow representation for each relationship type.
+     */
+    @Test
+    public void testGetArrowString(){
+        Assertions.assertEquals(" -----|> ", RelationshipType.GENERALIZATION.getArrowString());
+        Assertions.assertEquals(" <*>---- ", RelationshipType.COMPOSITION.getArrowString());
```
Comment (colbycrutcher) [inline src/test/java/RelationshipTypeTests.java:130]: tests look good for arrows

File: src/test/java/SaveManagerTests.java [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+    /**
+     * Tests saving custom UML class data and relationships to a JSON file.
+     *
+     * @throws IOException if an error occurs during file operations.
+     */
+    @Test
+    void testSaveToJSON_CustomData() throws IOException {
+        Map<String, UMLClassInfo> classMap = new HashMap<>();
+        classMap.put("Dog", new UMLClassInfo("Dog"));
+        classMap.put("Cat", new UMLClassInfo("Cat"));
```
Comment (colbycrutcher) [inline src/test/java/SaveManagerTests.java:185]: good savemanager tests

File: src/test/java/StateManagerTest.java [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+        stateManager.undo();
+        assertEquals(2, UMLClass.classMap.size(), "The class map should have two entries after the first undo.");
+        assertTrue(UMLClass.classMap.containsKey("Dog"), "The class map should contain 'Dog' after undo.");
+
+        stateManager.redo();
+        assertEquals(0, UMLClass.classMap.size(), "The class map should be empty after redo to the most recent state.");
+
+        stateManager.undo();
+        assertEquals(2, UMLClass.classMap.size(), "The class map should have two entries after the second undo.");
+        assertTrue(UMLClass.classMap.containsKey("Dog"), "The class map should contain 'Dog' after undo.");
```
Comment (colbycrutcher) [inline src/test/java/StateManagerTest.java:140]: good undo redo tests

File: src/test/java/UMLClassTests.java [hunk 5]
```diff
@@ -98,4 +133,169 @@ public void testRenamingNonExistingClass() {
...
+    }
+
+    /**
+     * Tests setting the position of a class using x and y coordinates.
+     */
+    @Test
+    public void testSetPosition() {
+        UMLClass.addClass("DOG");
+        UMLClassInfo info = UMLClass.getClassInfo("DOG");
+        info.setX(100);
```
Comment (colbycrutcher) [inline src/test/java/UMLClassTests.java:294]: good class tests

File: src/test/java/UMLMethodsTests.java [hunk 6]
```diff
@@ -220,4 +221,65 @@ public void testChangeAllParameters() {
...
+        UMLParameterInfo old1 = new UMLParameterInfo("boolean", "IsPuppy");
+        methodInfo.addParameter(old);
+        methodInfo.addParameter(old1);
+
+        UMLParameterInfo newParam = new UMLParameterInfo("string", "dogString");
+
+        methodInfo.changeParameter(old, newParam);
+
+        List<UMLParameterInfo> expected = Arrays.asList(newParam, old1);
+        assertEquals(expected, methodInfo.getParameters());
```
Comment (colbycrutcher) [inline src/test/java/UMLMethodsTests.java:281]: good method tests.

**Review by colbycrutcher — APPROVED** [review]
Looks good. Fixes some code that didn't pass tests. Tests look good as well

**Review by AaronWB — APPROVED** [review]
Tests are written well and are within the 90% Coverage Dr. Cain wants while also making sure that tested methods are properly working.

Good Job and Looks Good.

**Pull request merged**